sufia-models/app/services/sufia/generic_file_indexing_service.rb

```diff
@@ -13,6 +13,7 @@ def generate_solr_document
         # between files on disk (in fcrepo.binary-store-path) and objects
         # in the repository.
         solr_doc[Solrizer.solr_name('digest', :symbol)] = digest_from_content
+        object.index_collection_ids(solr_doc) unless Sufia.config.collection_facet.nil?
```

Comment:
This code does not work.  The fields added by index_collection_ids are not added to the solr doc when called in this way.

Also, there are at least two paths for adding a file to a collection and they don't both go through the CollectionControllerBehavior code.  When adding from Dashboard, this code is used such that the collection is causing the reindex.  When adding through Batch Upload, each generic file is indexed and the collection update code is never called.  So even if the proposed code worked, taking index_collection_ids out of the solr indexer code would prevent generic files from having the proper fields any time they are indexed from anywhere besides a collection update.